homeassistant/components/rflink.py

```diff
@@ -314,23 +332,62 @@ def is_on(self):
     @property
     def assumed_state(self):
         """Assume device state until first device event sets state."""
-        return self._state is STATE_UNKNOWN
+        return self._state is None
 
     @property
     def available(self):
         """Return True if entity is available."""
         return self._available
 
     @callback
-    def set_availability(self, availability):
+    def _availability_callback(self, availability):
         """Update availability state."""
         self._available = availability
         self.async_schedule_update_ha_state()
 
     async def async_added_to_hass(self):
         """Register update callback."""
+        # Remove temporary bogus entity_id if added
+        tmp_entity = TMP_ENTITY.format(self._device_id)
+        if tmp_entity in self.hass.data[DATA_ENTITY_LOOKUP][
+                EVENT_KEY_SENSOR][self._device_id]:
+            self.hass.data[DATA_ENTITY_LOOKUP][
+                EVENT_KEY_SENSOR][self._device_id].remove(tmp_entity)
+
+        # Register id and aliases
+        self.hass.data[DATA_ENTITY_LOOKUP][
+            EVENT_KEY_COMMAND][self._device_id].append(self.entity_id)
+        if self._group:
+            self.hass.data[DATA_ENTITY_GROUP_LOOKUP][
+                EVENT_KEY_COMMAND][self._device_id].append(self.entity_id)
+        # aliases respond to both normal and group commands (allon/alloff)
+        if self._aliases:
+            for _id in self._aliases:
+                self.hass.data[DATA_ENTITY_LOOKUP][
+                    EVENT_KEY_COMMAND][_id].append(self.entity_id)
+                self.hass.data[DATA_ENTITY_GROUP_LOOKUP][
+                    EVENT_KEY_COMMAND][_id].append(self.entity_id)
+        # group_aliases only respond to group commands (allon/alloff)
+        if self._group_aliases:
+            for _id in self._group_aliases:
+                self.hass.data[DATA_ENTITY_GROUP_LOOKUP][
+                    EVENT_KEY_COMMAND][_id].append(self.entity_id)
+        # nogroup_aliases only respond to normal commands
+        if self._nogroup_aliases:
+            for _id in self._nogroup_aliases:
+                self.hass.data[DATA_ENTITY_LOOKUP][
+                    EVENT_KEY_COMMAND][_id].append(self.entity_id)
         async_dispatcher_connect(self.hass, SIGNAL_AVAILABILITY,
-                                 self.set_availability)
+                                 self._availability_callback)
+        async_dispatcher_connect(self.hass,
+                                 SIGNAL_HANDLE_EVENT.format(self.entity_id),
+                                 self.handle_event_callback)
+
+        # Send signal to process the initial event after entity is created
+        if self._initial_event:
+            async_dispatcher_send(self.hass,
```

Comment:
Call the event handler method directly since we're inside the class here. Ie use `self.handle_event_callback`.